src/mlpack/methods/amf/init_rules/given_init.hpp

```diff
@@ -28,32 +28,79 @@ class GivenInitialization
 {
  public:
   // Empty constructor required for the InitializeRule template.
-  GivenInitialization() { }
+  GivenInitialization() : wIsGiven(false), hIsGiven(false) { }
 
   // Initialize the GivenInitialization object with the given matrices.
-  GivenInitialization(const arma::mat& w, const arma::mat& h) : w(w), h(h) { }
+  GivenInitialization(const arma::mat& w, const arma::mat& h) :
+    w(w), h(h), wIsGiven(true), hIsGiven(true) { }
 
   // Initialize the GivenInitialization object, taking control of the given
   // matrices.
   GivenInitialization(const arma::mat&& w, const arma::mat&& h) :
     w(std::move(w)),
-    h(std::move(h))
+    h(std::move(h)),
+    wIsGiven(true),
+    hIsGiven(true)
   { }
 
-  /**
-   * Fill W and H with random uniform noise.
-   *
-   * @param V Input matrix.
-   * @param r Rank of decomposition.
-   * @param W W matrix, to be filled with random noise.
-   * @param H H matrix, to be filled with random noise.
-   */
+  // Initialize either H or W with the given matrix.
+  GivenInitialization(const arma::mat& m, const bool whichMatrix = true)
+  {
+    if (whichMatrix)
+    {
+      w = m;
+      wIsGiven = true;
+      hIsGiven = false;
+    }
+    else
+    {
+      h = m;
+      wIsGiven = false;
+      hIsGiven = true;
+    }
+  }
+
+  // Initialize either H or W, taking control of the given matrix.
+  GivenInitialization(const arma::mat&& m, const bool whichMatrix = true)
+  {
+    if (whichMatrix)
+    {
+      w = std::move(m);
+      wIsGiven = true;
+      hIsGiven = false;
+    }
+    else
+    {
+      h = std::move(m);
+      wIsGiven = false;
+      hIsGiven = true;
+    }
+  }
+
+   /**
+    * Fill W and H with given matrices.
+    *
+    * @param V Input matrix.
+    * @param r Rank of decomposition.
+    * @param W W matrix, to be initialized to given matrix.
+    * @param H H matrix, to be initialized to given matrix.
+    */
```

Comment:
I think this is also indented one space too many (I can't exactly tell now, I'm looking at this on a phone in the back of a kind of bouncy car :)).

Reply:
Please refer to my response above.